src/error.rs

```diff
@@ -1,38 +1,50 @@
 use jsonpath_lib::JsonPathError;
+use std::fmt;
 
 #[derive(Debug)]
-pub struct Error {
-    pub msg: String,
+pub enum Error {
+    GeneralError(String),
+    PathNotAnObject,
+    WrongStaticPath,
 }
```

Comment:
Maybe it's time to switch to [thiserror](https://docs.rs/thiserror/1.0.19/thiserror/) (the de factor standard for structured errors) and remove some of the boilerplate in this file?